pkg/upgrade/handle_batch.go

```diff
@@ -54,6 +54,17 @@ func (ctl *Controller) handleJobs(ctx context.Context) error {
 		case err != nil:
 			return obj, err
 		}
+		if func(jobNode string, applyingNodes []string) bool {
+			for _, node := range applyingNodes {
+				if jobNode == node {
+					return false
+				}
+			}
+			return true
+		}(obj.Labels[upgradeapi.LabelNode], plan.Status.Applying) {
+			return obj, deleteJob(jobs, obj, metav1.DeletePropagationBackground)
```

Comment:
This will prematurely delete a job that has completed successfully (because the node for which it ran will drop off of the "applying" list).

I do not like the idea of deleting a job that that is neither failed nor completed but otherwise meets all of the plan's criteria. **But if we must prune here, because the node that the job is targeting IS NOT on the plan's `.status.applying` list, then it should be the last conditional action of this function.**

Additionally, given that there are no limits placed on the length of `.status.applying` we should eschew looping through all elements looking for a match and instead leverage the fact that the `.status.applying` is sorted which means we can use https://golang.org/pkg/sort/#SearchStrings or https://golang.org/pkg/sort/#Search to perform binary search.